internal/tiltfile/tiltfile_state.go

```diff
@@ -616,12 +622,59 @@ func (s *tiltfileState) assembleK8sV2() error {
 				r.name = opts.newName
 				s.k8sByName[r.name] = r
 			}
+
+			selectors := make([]k8sObjectSelector, len(opts.objects))
+			for i, o := range opts.objects {
+				s, err := selectorFromString(o)
+				if err != nil {
+					return errors.Wrapf(err, "Error making selector from string '%s'", o)
+				}
+				selectors[i] = s
+			}
+
+			claims := make(map[k8s.K8sEntity]claimPair)
+			for i, sel := range selectors {
+				for _, r := range s.k8s {
+					resourcedMatches := filterEntitiesBySelector(r.entities, sel)
+					if len(resourcedMatches) > 0 {
+						return fmt.Errorf("object %s already belongs to resource %s", opts.objects[i], r.name)
+					}
+				}
+
+				unresourcedMatches := filterEntitiesBySelector(s.k8sUnresourced, sel)
+				if len(unresourcedMatches) != 1 {
+					errMsg := fmt.Sprintf("Found %d matches for %s in remaining YAML. Object must match exactly 1 entity", len(unresourcedMatches), opts.objects[i])
```

Comment:
Maybe "selector is not unique, it matches each of these objects" but print full name